packages/core/src/ckb/transaction.ts

```diff
@@ -1856,6 +1899,15 @@ export class Transaction extends mol.Entity.Base<
     while (true) {
       const tx = this.clone();
```

Comment:
There is an issue right here: clone doesn't copy input metadata, so we are still re-fetching every single cell.

Let's say the current cell pool being examined is bigger that than the cache size, the cache will be thrashed and these requests will spill out to the underlying RPC. (Even worse if the Client doesn't have a cache)

While this is out of the scope of this FR, it could make sense to let the offline check operate on the original transaction, not on the clone.

What's your take on this @Hanssen?

Also, in general, what's preventing us from cloning and prevent re-fetching in general by then shallowly copying the inputs?

Reply:
> it could make sense to let the offline check operate on the original transaction, not on the clone.

For this, because the `change` function modifies the transaction, but we need to do this multiple times on the original transaction, we must clone the transaction to preserve the original version.